src/Illuminate/Support/helpers.php

```diff
@@ -210,6 +210,37 @@ function preg_replace_array($pattern, array $replacements, $subject)
     }
 }
 
+if (! function_exists('rethrow')) {
+    /**
+     * Return callback result, but convert to specified exception when exception thrown.
+     *
+     * @param  callable  $callback
+     * @param  \Throwable|string|callable  $exception
+     * @param  mixed  ...$parameters
+     * @return mixed
+     *
+     * @throws \Throwable
+     */
+    function rethrow(callable $callback, $exception, ...$parameters)
+    {
+        try {
+            return $callback();
+        } catch (Exception $e) {
+            if (is_callable($exception)) {
+                throw $exception($e);
+            }
+
+            if (is_string($exception) && class_exists($exception)) {
+                $exception = $parameters
+                    ? new $exception(...$parameters)
+                    : new $exception($e->getMessage(), $e->getCode());
```

Comment:
setting the code like that will crash if it's a PDO exception because getCode can return a string but the exception constuctor only accepts ints.

Reply:
good catch, thanks, wasn’t aware. In the event the PR will be reopened I’ll be happy to make further changes to cover this.